classes/notification/PKPNotificationOperationManager.inc.php

```diff
@@ -144,11 +144,11 @@ public function getParamsForCurrentLocale($params) {
 	 * @param $level int
 	 * @param $params array
 	 * @param $suppressEmail boolean Whether or not to suppress the notification email.
+	 * @param $mailConfigurator callable Enables the customization of the Notification email
 	 * @return Notification object
 	 */
-	public function createNotification($request, $userId = null, $notificationType, $contextId = null, $assocType = null, $assocId = null, $level = NOTIFICATION_LEVEL_NORMAL, $params = null, $suppressEmail = false) {
+	public function createNotification(Request $request, $userId = null, $notificationType, $contextId = null, $assocType = null, $assocId = null, $level = NOTIFICATION_LEVEL_NORMAL, $params = null, $suppressEmail = false, callable $mailConfigurator = null) {
```

Comment:
In order to customize the notification email, I was going to overwrite the getMailTemplate function, but the [$mail->assignParams](https://github.com/pkp/pkp-lib/pull/5005/commits/a9b8f22c06ddba880087cb2a9e3208f63c9ca0a7#diff-fbb83136a570e8ded8ae235091b6d849R394) didn't allowed me to continue. So I've added this extra parameter to provide a last chance to do something with the email.

Reply:
By now I think it's ok too, but such methods with many arguments are good targets for a refactoring.